### ISSUE TITLE
Multiple node pools - z2jh labeling and taints

### ISSUE THREAD

**Issue by consideRatio** [jupyterhub/mybinder.org-deploy#672]
TL;DR: add `hub.jupyter.org/node-purpose: user` or `hub.jupyter.org/node-purpose: core` label to the new node pools you create. 

In https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/758 meaning is given to the node label `hub.jupyter.org/node-purpose` and its values `core` and `user` that the hub/proxy/singleuser pods can with a configurable value `ignore`, `prefer` or `require` to schedule on using the `nodeAffinity` mechanism behind the scenes. If you add such label in advance, you don't need to recreate the node pool to reap the benefits of the z2jh solution later.

To really do well with autoscaling a user node-pool, I'd suggest one also applies a taint to those node-pools and a corresponding toleration. If not, pods like kube-dns may end up there and block, or a proxy pod that were restarting during an upgrade but lacked resources on a core node for a brief moment and scheduled on a user node. This can wait until https://github.com/jupyterhub/kubespawner/pull/205 lands.

**About `nodeAffinity`**: it is the newer more flexible option to `nodeSelector` that can only do the `require` option and can not declare "for a node having one of these labels" but instead only "for nodes having this label".
